src/components/SelectionList/BaseListItem.tsx

```diff
@@ -43,6 +44,14 @@ function BaseListItem<TItem extends ListItem>({
         return rightHandSideComponent;
     };
 
+    const handleCheckboxPress = () => {
```

Comment:
use `useCallback` with `item` dependency.

or I think `onPress={() => (onCheckboxPress ?? onSelectRow)(item)}` is better without defining this function